### PR TITLE
Update NuGetCommand@2 with workaround for microsoft ADO

### PR DESCRIPTION
**Pull request by jonthysell**
The microsoft ADO instance maintains two different tasks named NuGetCommand. This means pipelines running on that ADO instance need the unique GUID to identify it.

Part of #5540.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7238)

### CODE REVIEW

File: .ado/templates/build-rnw.yml [hunk 1]
```diff
@@ -31,7 +31,8 @@ steps:
       installVsComponents: ${{ parameters.installVsComponents }}
       debug: ${{ parameters.debug }}
 
-  - task: NuGetCommand@2
+  # NuGetCommand@2 workaround: https://developercommunity.visualstudio.com/content/problem/288534/vsts-yaml-build-failure-the-task-name-nugetcommand.html
+  - task: 333b11bd-d341-40d9-afcf-b32d5ce6f23b@2
```
Comment (asklar) [inline .ado/templates/build-rnw.yml:35]: i wonder if this is even required now that we can do `/restore /p:RestorePackagesConfig=true` ?

**Review by asklar — APPROVED** [review]

File: .ado/templates/build-rnw.yml (hunk 1 above)
Comment (jonthysell) [reply .ado/templates/build-rnw.yml:35]: The issue was brought up on #6958, but I'm not taking on a further refactor for the purposes of getting codesigning unblocked.

**Review by dannyvv — APPROVED** [review]

**Comment by jonthysell** [comment]
I don't know why CG keeps randomly failing during the day (but not at night), but I need to get this in.

**Pull request merged**